source/particle/interpolator/cell_average.cc

```diff
@@ -57,8 +64,18 @@ namespace aspect
           found_cell = cell;
 
         const types::LevelInd cell_index = std::make_pair(found_cell->level(),found_cell->index());
-        const std::pair<typename std::multimap<types::LevelInd, Particle<dim> >::const_iterator,
-              typename std::multimap<types::LevelInd, Particle<dim> >::const_iterator> particle_range = particles.equal_range(cell_index);
+
+        std::pair<typename std::multimap<types::LevelInd, Particle<dim> >::const_iterator,
+            typename std::multimap<types::LevelInd, Particle<dim> >::const_iterator> particle_range;
+
+        if (cell->is_locally_owned())
+          {
+            particle_range = particles.equal_range(cell_index);
```

Comment:
I think you can write this more elegantly as
```
  std::pair<...> particle_range = (cell->is_locally_owned()
                                                   ?
                                                   particles.equal_range(cell_index)
                                                   :
                                                   ghost_particles.equal_range(cell_index));
```
This way you could also make it `const`.